File: java-selenium/sauce-demo-automation/src/test/java/com/saucedemo/automation/pages/ProductDetailsPage.java
package com.saucedemo.automation.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class ProductDetailsPage {
    private WebDriver driver;

    @FindBy(className = "inventory_details_name")
    private WebElement productName;

    @FindBy(className = "inventory_details_price")
    private WebElement productPrice;

    @FindBy(className = "inventory_details_desc")
    private WebElement productDescription;

    @FindBy(className = "btn_inventory")
    private WebElement addToCartButton;

    public ProductDetailsPage(WebDriver driver) {
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }

    public String getProductName() {
        return productName.getText();
    }

    public String getProductPrice() {
        return productPrice.getText();
    }

    public String getProductDescription() {
        return productDescription.getText();
    }

    public void clickAddToCart() {
        addToCartButton.click();
    }

    public String getCurrentUrl() {
        return driver.getCurrentUrl();
    }
}

File: cypress/sauce-demo-automation/cypress/support/pages/ProductDetailsPage.js
class ProductDetailsPage {
    get productName() {
        return cy.get('.inventory_details_name');
    }

    get productPrice() {
        return cy.get('.inventory_details_price');
    }

    get productDescription() {
        return cy.get('.inventory_details_desc');
    }

    get addToCartButton() {
        return cy.get('.btn_inventory');
    }

    getProductName() {
        return this.productName.invoke('text');
    }

    getProductPrice() {
        return this.productPrice.invoke('text');
    }

    getProductDescription() {
        return this.productDescription.invoke('text');
    }

    clickAddToCart() {
        this.addToCartButton.click();
    }
}

export default new ProductDetailsPage();
